taf/updater/updater_pipeline.py

```diff
@@ -557,18 +578,25 @@ def check_if_local_target_repositories_clean(self):
             self.state.event = Event.FAILED
             return UpdateStatus.FAILED
 
-    @log_on_start(
-        INFO, "Cloning target repositories which are not on disk...", logger=taf_logger
-    )
+    @log_on_start(DEBUG, "Cloning target repositories to temp...", logger=taf_logger)
     @log_on_end(INFO, "Finished cloning target repositories", logger=taf_logger)
-    def clone_target_repositories_if_not_on_disk(self):
+    def clone_target_repositories_to_temp(self):
         try:
-            self.state.cloned_target_repositories = []
-            for repository in self.state.target_repositories.values():
-                is_git_repository = repository.is_git_repository_root
-                if not is_git_repository:
-                    repository.clone(no_checkout=True)
-                    self.state.cloned_target_repositories.append(repository)
+            self.state.repos_on_disk = []
+            self.state.repos_not_on_disk = []
+            for temp_repo in self.state.temp_target_repositories.values():
+                users_repo = self.state.users_target_repositories[temp_repo.name]
+                is_git_repository = users_repo.is_git_repository_root
+                if is_git_repository:
+                    temp_repo.clone_from_disk(
+                        users_repo.path,
+                        users_repo.get_remote_url(),
+                        is_bare_repository=True,
+                    )
```

Comment:
question: Will this do a complete clone of the target repositories to temp dir? I can imagine a set of target repositories having gigabytes of data, (maybe of tens of gigabytes, if cloning a top-level stele). Is this approach of a full local clone going to scale if the root authentication repository references over 10-20 other authentication repos?

Reply:
That did cross my mind. I added except `KeyboardInterrupt`. Need to figure out how to integrate with codify